components/managed-identity/managed-identity.tf

```diff
@@ -140,6 +140,14 @@ resource "azurerm_role_assignment" "service_operator" {
   role_definition_name = "Contributor"
   scope                = data.azurerm_subscription.subscription.id
 }
+
+resource "azurerm_role_assignment" "preview_mi" {
+  count                = var.env == "preview" ? 1 : 0
+  principal_id         = data.azurerm_user_assigned_identity.aks.principal_id
+  scope                = "/subscriptions/1c4f0704-a29e-403d-b719-b90c34ef14c9"
```

Comment:
Could we also use this where it's already defined similar to ${local.mi_cft[var.env].scope} ?